DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc

```diff
@@ -1609,14 +1697,32 @@ void Converter<DDLShapeless>::operator()(xml_h element) const {
 
   printout(ns.context()->debug_shapes ? ALWAYS : DEBUG,
            "DD4CMS",
-           "+   Shapeless: THIS ONE CAN ONLY BE USED AT THE VOLUME LEVEL -> Assembly%s",
+           "+   Shapeless: THIS ONE CAN ONLY BE USED AT THE VOLUME LEVEL -> Shapeless: %s",
            nam.c_str());
 
 #endif
 
   ns.addSolid(nam, Box(1, 1, 1));
 }
 
+/// Converter for <Assembly/> tags
+template <>
+void Converter<DDLAssembly>::operator()(xml_h element) const {
+  cms::DDNamespace ns(_param<cms::DDParsingContext>());
+  xml_dim_t e(element);
+  string nam = e.nameStr();
+
+#ifdef EDM_ML_DEBUG
+  printout(ns.context()->debug_shapes ? ALWAYS : DEBUG,
+           "DD4CMS",
+           "+   Assembly: Adding solid -> Assembly: %s",
+           nam.c_str());
+#endif
+
+  ns.addSolid(nam, Box(nam, 1, 1, 1));  // Add a dummy solid to allow assembly to be treated like other volumes
+  ns.addAssemblySolid(nam);
```

Comment:
@cvuosalo - same here, please, let dd4hep handle it. Assembly is a shapeless solid.

Reply:
Line 1723 is bookkeeping to keep track of all the Assembly names in the Solid Section. Adding the Box is probably not necessary, but I put it in to be safe.